R/size.R

```diff
@@ -99,43 +99,49 @@ vec_size2 <- function(nx, ny) {
 #'
 #' vec_slice(mtcars, 1:3)
 vec_slice <- function(x, i) {
-  i <- get_slice_index(i, x)
+  .Call(vctrs_slice, x, maybe_missing(i))
```

Comment:
Do you want to make add `UseMethod("vec_slice")` now to make it more consistent with the other internal generics?

Reply:
I think we would need a `dispatch` argument so the default method can call back `vec_slice()`with `dispatch = FALSE`. This will be also useful for implementing `[` with `vec_slice()` semantics.

In general it seems a good principle to allow the caller to turn off dispatch. We seem to rely quite a lot on `.subset()` and `.subset2()` to manipulate the underlying R type, this would be useful in vctrs as well. Best implemented in a future PR I think.

Reply:
Hmmmm, I think I would rather have a separate function for that. Otherwise the methods will also need to have `dispatch = FALSE` argument, which seems weird/confusing/wrong.

Reply:
Makes sense. The recursion is a bit tricky so best left for another PR.